### ISSUE TITLE
Leaf - N-Connecta 2022 (UK)

### ISSUE THREAD

**Issue by LeeUKM** [dan-r/HomeAssistant-NissanConnect#33]
Just linked our 2022 Leaf N-Connect and works perfectly.

THANK YOU very much for this excellent tool.

One annoyance with the LEAF is being unable to set 80-85% charge stop to protect the battery. I can now create Automations to start charging at off-peak and stop at 85%.

**Comment by dan-r** [comment]
Glad its working well for you.
How are you stopping the charge? One major limitation with the Nissan API and app is that there's no pause/stop charge function. Personally I have an Ohme charger and stop it through this.

**Comment by LeeUKM** [comment]
Your add-in is one of the last key parts of the puzzle.

I owe you a coffee..

By reading the Leaf battery level prior to charge commence, our Leaf if 40kw, therefore 10% is 4kwh of charge. 

Example:

Battery SOC - 55%
So a helper calculated 85% - 55% = 30%
Multiply 30 by 0.4 to give 12kwh.

Using the Zappi Integration, I have a manual Boost Automation which I can slide from 1 to 36. At midnight, an automation is triggered then sets the Zappi to Boost by whatever amount was set.

By creating an automation that triggers when the car is plugged in, this automation sets the amount of Boost needed from midnight to bring to 85%.

I could monitor the battery SOC using your add-in but to avoid overshooting due to Internet or other issue the SOC isn't updated, I set the amount ar start.

I can try posting the automations and helpers if they help.

So basically

Get home and pluggcar in HA detects plug in
Hone Assistant reads EV SOC and calculates amount
At midnight this is sent to the EV to start charge
ZAPPI stops once Boost amount reached

This will give the battery a long and happy life.

Hope this helps..

**Comment by LeeUKM** [comment]
Just to expand on this:

I assemble Lithium Batteries for domestic Solar and off-peak storage and know plenty on longevity of Battery Cells.

Many manufacturers of EVs are now building in "Battery saver" settings which is stop charging at 80% although this can be overridden.

I was playing last night and made the target value Variable. Ideally a battery likes to sit around 50% and not below 20%. Therefore is possible to add just enough to commute with some reserve and over Winter, likely set ours to 65%.

The absolute worst thing for an EV is nightly charging from and already high level to 100% every night. At 65%, ours will last 3 days to 20% - so every 3 days it will be charged back to 65%.

Every two weeks, I'll run a nighttime 100% charge to balance the Cells in the battery.

On the dashboard- I have "Solar Charge", "Night Charge 100%" and "Limit Charge" and depending on which is selected- the Zappi will run as told.
